src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2195,6 +2199,30 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
             return method?.UseUpdatedEscapeRules == true;
         }
 
+        private static bool ShouldInferDeclarationExpressionValEscape(BoundExpression argument, [NotNullWhen(true)] out SourceLocalSymbol? localSymbol)
+        {
+            var symbol = argument switch
+            {
+                BoundDeconstructValuePlaceholder p => p.ExpressionSymbol,
+                BoundLocal { DeclarationKind: not BoundLocalDeclarationKind.None } l => l.ExpressionSymbol,
+                _ => null
+            };
+            if (symbol is SourceLocalSymbol { ValEscapeScope: CallingMethodScope } local)
```

Comment:
note that we really would prefer to use a sentinel ValEscapeScope in cases like these (e.g. with value `1 << 32`.). I attempted to do this but it started to really churn and wanted to be expedient for the purposes of getting this fix in.